molgenis-data-postgresql/src/main/java/org/molgenis/data/postgresql/PostgreSqlExceptionTranslator.java

```diff
@@ -201,8 +201,22 @@ MolgenisValidationException translateForeignKeyViolation(PSQLException pSqlExcep
 			throw new RuntimeException("Error translating exception", pSqlException);
 		}
 		String value = m.group(1);
+
+		String constraintViolationMessageTemplate;
+		if (detailMessage.contains("still referenced from"))
+		{
+			// ERROR: update or delete on table "x" violates foreign key constraint "y" on table "z"
+			// Detail: Key (k)=(v) is still referenced from table "x".
+			constraintViolationMessageTemplate = "Value '%s' for attribute '%s' is referenced by entity '%s'.";
```

Comment:
I'd expect something like "Entity row with value '%2$s' = '%1$s' is referenced by entity '%3$s'.".